### ISSUE TITLE
Panic `byte index 18 is out of bounds of` in `crates/oxc_span/src/span/mod.rs`

### ISSUE THREAD

**Issue by qarmin** [oxc-project/oxc#5434]
File content(at the bottom should be attached raw, not formatted file - github removes some non-printable characters, so copying from here may not work):
```
switch(n){case'':/s/}
```

command
```
timeout -v 100 oxlint -D all -D nursery --import-plugin --jsdoc-plugin --jest-plugin --vitest-plugin --jsx-a11y-plugin --nextjs-plugin --react-perf-plugin TEST___FILE.js --fix
```

cause this
```
thread '<unnamed>' panicked at crates/oxc_span/src/span/mod.rs:315:21:
byte index 18 is out of bounds of ``
stack backtrace:
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

##### Automatic Fuzzer note, output status "None", output signal "Some(6)"
```

Looks like quite new panic, introduced ~1/2 day ago

[compressed.zip](https://github.com/user-attachments/files/16862063/compressed.zip)

**Comment by rzvxa** [comment]
It will be fixed via #5431

**Comment by rzvxa** [comment]
Fixed via #5431
